docs/user-guide/installation/helics_cmake_options.md

```diff
@@ -73,6 +73,19 @@ Options effect the connection of libraries used in HELICS and how they are linke
 - `HELICS_ENABLE_SUBPROJECT_TESTS`: \[Default=OFF\] Turn on some additional tests for using HELICS as a subproject, mainly used in some of the CI testing to make sure HELICS works as a subproject.
 - `HELICS_ENABLE_CLANG_TOOLS`: \[Default=OFF\] Enables some helper targets for using clang-tidy and clang-format.
 
+#### Options related to using external/vendored libraries
+
+These options are for making HELICS look for an external installation or system copy of libraries instead of using the vendored copy included with HELICS. The search process for locating the external copy of libraries follows the typical [CMake Config Mode Search Procedure](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure). Most users should leave these options `OFF`.
+
+When enabling these options, if the static library variant is found it _must_ have been compiled using the option `CMAKE_POSITION_INDEPENDENT_CODE=ON` or _you will encounter linking errors_.
+
+Furthermore, there are no guarantees that HELICS will compile using arbitrary versions of these libraries other than the exact version that is included as a vendored library in the HELICS repository.
+
```

Comment:
Should we add any comments about which versions of these libraries will likely work?

Reply:
Are we testing any of this in any CI builds (likely not)  and should we be?

Reply:
I don't want to list any exact versions, since as I found while testing the feature, the version in development of `fmtlib` that comes after 8.1.1 (the tagged version corresponding to the current submodule git commit hash) results in compiler errors about `const` being lost when using `this`.

I added more information on getting what the current commit hashes for submodules are; hopefully the use cases for these options are fairly advanced scenarios, so users should already have some idea how to troubleshoot issues that arise from conflicting versions.

Reply:
These aren't tested in any CI builds. Some testing should eventually be figured out, but it needs to use an install of the exact same version that we vendor or we might encounter situations where a minor change in a new version breaks the external library build but not the vendored build (or the other way around).